app/utils/guaranteeParsing/RouteChecker.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.guaranteeParsing
+
+import com.google.inject.Inject
+import models.ParseError
+import models.ParseError.InappropriateDepartureOffice
+
+import scala.xml.NodeSeq
+
+class RouteChecker @Inject()(xmlReaders: GuaranteeXmlReaders) {
+
+  def gbOnlyCheck(xml: NodeSeq): Either[ParseError, Boolean] =
+    xmlReaders.officeOfDeparture(xml) match {
+      case Left(error) => Left(error)
+      case Right(departure) => {
+        xmlReaders.officeOfDestination(xml) match {
+          case Left(error) => Left(error)
+          case Right(destination) => {
+            (departure.value.startsWith("GB"), destination.value.startsWith("GB")) match {
+              case (true, destinationGB) => Right(destinationGB)
+              case (false, _) => Left(InappropriateDepartureOffice("Inappropriate Departure Office"))
```

Comment:
An if statement would look clearer here.

`if (departure.value.startsWith("GB") && destination.value.startsWith("GB")) Right(true) else Left(InappropriateDepartureOffice("Inappropriate Departure Office"))`

Does the check need to be case sensitive?

Could the method signature be changed to return a Boolean? With ParseError created higher up the call stack. The method name could also be changed to `isGBOnly`.

Reply:
I need to use the value of destination.value.startsWith so that if statement wouldn't work, but another one would.

No, I need the ParseError Either because the two mapped xmlReader methods both return ParseErrors and those values need to be handled